### PR TITLE
tee: move from getopts to clap

### LINKED ISSUE #1793: tee: move from getopts to clap
Code is in:
https://github.com/uutils/coreutils/blob/master/src/uu/tee/src/tee.rs

Tests:
https://github.com/uutils/coreutils/blob/master/tests/by-util/test_tee.rs (need to add tests)

As example:
https://github.com/uutils/coreutils/commit/9081e120d6b866c94679d9e29c7bdb17ccccb4d0

### PR DESCRIPTION
**Pull request by jhscheer**
This closes  #1793

### CODE REVIEW

File: src/uu/tee/src/tee.rs [hunk 1]
```diff
@@ -5,88 +5,73 @@
...
-            Options {
-                program: NAME.to_owned(),
-                append: m.opt_present("append"),
-                ignore_interrupts: m.opt_present("ignore-interrupts"),
-                print_and_exit: to_print,
-                files: names,
-            }
-        })
-        .map_err(|message| warn(format!("{}", message).as_ref()))
+mod options {
```
Comment (sylvestre) [inline src/uu/tee/src/tee.rs:67]: could you please move it above the main ?
(for consistency with other programs)
thanks

File: src/uu/tee/src/tee.rs [hunk 2]
```diff
@@ -209,6 +194,6 @@ impl Read for NamedReader {
 }
 
 fn warn(message: &str) -> Error {
-    eprintln!("{}: {}", NAME, message);
-    Error::new(ErrorKind::Other, format!("{}: {}", NAME, message))
+    eprintln!("{}: {}", executable!(), message);
```
Comment (sylvestre) [inline src/uu/tee/src/tee.rs:197]: I wonder if we could not use the show_warning function here instead?
https://github.com/uutils/coreutils/blob/d00356eab0ea98bab7ed5b0aa1e262673f3a9e3c/src/uucore/src/lib/macros.rs#L30

**Review by sylvestre — CHANGES_REQUESTED** [review]
looks great, just some minor stuff

**Comment by sylvestre** [comment]
and please fix the clippy warning

File: src/uu/tee/src/tee.rs (hunk 1 above)
Comment (jhscheer) [reply src/uu/tee/src/tee.rs]: done

File: src/uu/tee/src/tee.rs (hunk 2 above)
Comment (jhscheer) [reply src/uu/tee/src/tee.rs]: done

**Comment by jhscheer** [comment]
> and please fix the clippy warning

The only clippy 0.0.212 (cb75ad5d 2021-02-10) warning I get is:
`86: inactive-code: code is inactive due to #[cfg] directives: unix is enabled`
which looks fine to me.

File: src/uu/tee/src/tee.rs [hunk 3]
```diff
@@ -209,6 +194,6 @@ impl Read for NamedReader {
 }
 
 fn warn(message: &str) -> Error {
-    eprintln!("{}: {}", NAME, message);
-    Error::new(ErrorKind::Other, format!("{}: {}", NAME, message))
+    show_warning!("{}", message);
```
Comment (sylvestre) [inline src/uu/tee/src/tee.rs:197]: I was talking about replacing call to warn() by show_warning() 
and remove this function from this file :)

Comment (jhscheer) [reply src/uu/tee/src/tee.rs:197]: Okay, but what about the `Error::new` ? That would be missing if warn() gets replaced.
Also do happen to know how I could provoke an error to test the output?

Comment (jhscheer) [reply src/uu/tee/src/tee.rs]: done

**Pull request merged**